Presentation.Web/app/helpers/ExcelExportHelper.ts

```diff
@@ -155,5 +155,109 @@
+         * @param entry
+         * @param scope
+         * @param toolbar
+         */
+        static setupExcelExportDropdown(excelConfig: () => Models.IExcelConfig, mainGrid: () => IKendoGrid<any>, scope: ng.IScope, toolbar: IKendoGridToolbarItem[]) {
+            const entry = this.createExcelExportDropdownEntry(excelConfig, mainGrid);
+            this.addExcelExportDropdownToToolbar(toolbar, entry);
+            this.setupKendoVm(scope, entry);
+        }
+        
+        /**
+          * Creates an object of type IKendoToolbarEntry configured as a dropdown for excel export
+         * @param excelConfig
+         * @param mainGrid
+         */
+        static createExcelExportDropdownEntry(excelConfig: () => Models.IExcelConfig, mainGrid: () => IKendoGrid<any>): Utility.KendoGrid.IKendoToolbarEntry {
+            return {
+                show: true,
+                id: Constants.ExcelExportDropdown.Id,
+                title: Constants.ExcelExportDropdown.DefaultTitle,
+                color: Utility.KendoGrid.KendoToolbarButtonColor.Grey,
+                position: Utility.KendoGrid.KendoToolbarButtonPosition.Right,
+                implementation: Utility.KendoGrid.KendoToolbarImplementation.DropDownList,
+                enabled: () => true,
+                dropDownConfiguration: {
+                    selectedOptionChanged: newItem => {
+                        if (newItem === null)
+                            return;
+
+                        const config = excelConfig();
+                        config.onlyVisibleColumns = false;
+
+                        if (newItem.id === Constants.ExcelExportDropdown.SelectOnlyVisibleId) {
+                            config.onlyVisibleColumns = true;
+                        }
+
+                        mainGrid().saveAsExcel();
+
+                        jQuery(`#${Constants.ExcelExportDropdown.Id}`).data(Constants.ExcelExportDropdown.DataKey)
+                            .value(Constants.ExcelExportDropdown.DefaultValue);
+                    },
+                    availableOptions: [
+                        {
+                            id: Constants.ExcelExportDropdown.SelectAllId,
+                            text: Constants.ExcelExportDropdown.SelectAllValue
+                        },
+                        {
+                            id: Constants.ExcelExportDropdown.SelectOnlyVisibleId,
+                            text: Constants.ExcelExportDropdown.SelectOnlyVisibleValue
+                        }]
+                }
+            }
+        }
+
+        /**
+         * Configures a specified entry to be a Dropdown with classes required for excel export to work,
+         * Adds the entry to the selected toolbar
+         * @param toolbar
+         * @param entry
+         */
+        static addExcelExportDropdownToToolbar(toolbar: IKendoGridToolbarItem[], entry: Utility.KendoGrid.IKendoToolbarEntry) {
```

Comment:
should all of these helpers be public or should some of them be private and not clutter the available methods?